tree/treeplayer/inc/ROOT/TDFInterface.hxx

```diff
@@ -1561,6 +1561,53 @@ public:
+   /// \param[in] merger A callable with signature `U(U,U)` or `void(std::vector<U>&)` used to merge the results of the accumulations of each thread
+   /// \param[in] columnName The column to be accumulated. If omitted, the first default column is used instead.
+   /// \param[in] initValue The accumulator variable is initialised to this value (or is default-constructed if the parameter is omitted)
+   ///
+   /// An accumulation takes two values, an accumulator variable and a column value. The accumulator variable is
+   /// initialized to initValue or default-constructed if initValue is omitted.
+   /// This action then calls the accumulator for each processed entry, passing in the accumulator variable and 
+   /// the current value of the column columnName.
+   /// If the signature is `U(U,T)` the accumulator variable is then copy-assigned the result of the execution of the callable.
+   /// Otherwise the signature of accumulator must be `void(U,T)`.
+   ///
+   /// The merger callable is used to merge the partial accumulation results of each processing thread. It is only called in multi-thread executions.
+   ///
+   /// This action is *lazy*: upon invocation of this method the calculation is
+   /// booked but not executed. See TResultProxy documentation.
+   template <typename AccFun, typename MergeFun, typename R = typename TTraits::CallableTraits<AccFun>::ret_type,
+             typename ArgTypes = typename TTraits::CallableTraits<AccFun>::arg_types,
+             typename U = TTraits::TakeFirstParameter_t<ArgTypes>,
+             typename T = TTraits::TakeFirstParameter_t<TTraits::RemoveFirstParameter_t<ArgTypes>>>
+   TResultProxy<U>
+   Accumulate(AccFun accumulator, MergeFun merger, std::string_view columnName = "", const U &initValue = U())
+   {
+      TDFInternal::CheckAccumulate<R, MergeFun>(ArgTypes());
+      auto loopManager = GetDataFrameChecked();
+      const auto columns = columnName.empty() ? ColumnNames_t() : ColumnNames_t({std::string(columnName)});
+      constexpr auto nColumns = ArgTypes::list_size;
+      const auto validColumnNames =
+         TDFInternal::GetValidatedColumnNames(*loopManager, 1, columns, fValidCustomColumns, fDataSource);
+      if (fDataSource)
+         TDFInternal::DefineDataSourceColumns(validColumnNames, *loopManager, TDFInternal::GenStaticSeq_t<nColumns>(),
+                                              ArgTypes(), *fDataSource);
+      auto accObjPtr = std::make_shared<U>(initValue);
+      using Helper_t = TDFInternal::AccumulateHelper<AccFun, MergeFun, R, U, T>;
+      using Action_t = typename TDFInternal::TAction<Helper_t, Proxied>;
+      auto action = std::make_shared<Action_t>(
+         Helper_t(std::move(accumulator), std::move(merger), accObjPtr, fProxiedPtr->GetNSlots()), validColumnNames,
+         *fProxiedPtr);
+      loopManager->Book(action);
+      return MakeResultProxy(accObjPtr, loopManager, action.get());
+   }
+
```

Comment:
Also after our discussion: in a second step after this PR, define Reduce in terms of Accumulate